src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,107 @@
+package duke;
+
+import java.util.Scanner;
+
+public class Ui {
+    private static String logo =
+            " ___________________\n" +
+                    ":' ,__________,  ':  `.\n" +
+                    "| '            `  |    `.\n" +
+                    "| |  Dukey     |  |      `.\n" +
+                    "| |            |  |        \\\n" +
+                    "| |  Hello!!!  |  |         ]\n" +
+                    "| |            |  |~~~~~~.  )\n" +
+                    "| `,__________,'  |\\__O\\_| ,'\n" +
+                    "|    _______      |     \\.`\n" +
+                    "|<> [___=___](@)<>|    .'\\\n" +
+                    "':________________/__.'   )\n" +
+                    "   (____________)        /\n" +
+                    "                        /\n" +
+                    "              _________/\n" +
+                    "  ___________/______\n" +
+                    " /''''=========='(@)\\___\n" +
+                    " |[][][][][][][][][]|   \\ _______\n" +
+                    " |[][][][][][][][][]|    \\__     \\\n" +
+                    " |[][][][][][][][][]|    |  \\..  |\n" +
+                    " \\------------------/    | ( # ) |\n" +
+                    "                         |  '''  |\n" +
+                    "                         \\_______/";
+    ;
+
+    /**
+     * Reads a command input from the user.
+     *
+     * @return The user-entered command.
+     */
+    public String readCommand() {
+        Scanner scanner = new Scanner(System.in);
+        return scanner.nextLine();
+    }
+
+    /**
+     * Displays a welcome message to the user upon starting the application.
+     */
+    public void showWelcomeMessage() {
+        System.out.println(logo);
+        System.out.println("Hello! I am Dukey.\n"
+                + "What can I do for you?");
+    }
+
+    /**
+     * Displays a line separator.
+     */
+    public void showLine() {
+        System.out.println("_______");
+    }
+
+    /**
+     * Displays a message after marking or unmarking a task, indicating the task's new status.
+     *
+     * @param taskMarked The task that was marked or unmarked.
+     * @param isMark     Whether the task was marked (true) or unmarked (false).
+     */
+    public void showMarkTaskMessage(String taskMarked, Boolean isMark) {
```

Comment:
Maybe for isMark should be renamed to isMarked?

Reply:
Good catch! Thanks!